pkg/serviceaccount/tokens_controller.go

```diff
@@ -307,9 +308,17 @@ func (e *TokensController) createSecret(serviceAccount *api.ServiceAccount) erro
 
 	_, err = serviceAccounts.Update(serviceAccount)
```

Comment:
The cleanup and conflict tolerance looks fine.

Looking closer, we also need to make sure the live service account we `Get` in order to `Update` is the same as the one we based our decision to create a secret on (either same resource version or identical `Secrets` list). If it's different, we can just bail and wait for the next update notification.

I'd like to do that check prior to creating the secret, just to save a potential reason for cleanup. That would involve unit test changes which expect client calls in a particular order. If you'd rather not mess with it, I can pick it up